site/content/features/jep445.md

```diff
@@ -0,0 +1,196 @@
+
+Ever since Java 1.0, the entry point for a Java program is a method 
+
+```
+public static void main(String[] args)
+```
+
+For teaching Java, this is not wonderful. Why `static`? Why `String[] args`? Many years ago, I submitted a patch to the `java` launcher that allowed it to instantiate any class without a `public static void main(String[] args)` and with a no-args constructor:
+
+```
+public class MyFirstObject { // My proposal—got nowhere
+   public MyFirstObject() {
+      System.out.println("Hello, world! I was just constructed.");
+   }
+}
+```
+
+It was rejected for some spurious reason.
+
+As an aside, for some time, it was possible to do something similar (but also not wonderful for teaching):
+
+```
+public class NoMain { // Up to Java 6
+   static {
+      System.out.println("Hello, world! Ignore the message below.");
+   }
+}
+```
+
+Calling `java NoMain` would execute the static initializer, followed by an error message about a missing `main`. This was fixed in Java 7
+
+In Java 8, the `java` launcher gained [special support](https://docs.oracle.com/javase/8/javafx/api/javafx/application/Application.html) for launching JavaFX applications without `public static void main`. But it doesn't help for teaching the “first cup of Java”.
+
+Finally, Java 21 will support easier ways for writing simpler Java programs. Professional programmers won't care, but it should lower the barrier to entry for beginning students.
+
+## Main Methods
+
+JEP 445 allows, as a preview feature in Java 21, a greater variety of `main` methods as an entry point of a Java program.
+
+* `main` need not be `static`, provided the class has a no-arg constructor
+* `main` need not be `public`. It can also have default visibility (no modifier), or be declared as `protected`. It cannot be `private`.
+* The `String[]` parameter can be omitted.
+
+That means there are twelve possible forms of `main`. Here is the shortest version of the classic “Hello, World!” program:
+
+```
+class HelloWorld {
+   void main() {
+      System.out.println("Hello, World!");
+   }
+}
+```
+
+You can launch a class that has multiple `main` methods—in fact, up to four, since you cannot overload on visibility. A `static main` method has preference over an instance method. If that leaves two candidates, the one with a `String[]` parameter has precedence.
+
+The `main` method can be declared in a superclass. This has always been true for `public static void main(String[])`. I don't see why anyone would do that, though.
```

Comment:
Possible use case: If you have a set similar tools that e.g. require the same setup (like setting up a database connection) you can now use the template method pattern to declare executable implementation classes of the different tools with a common base class having the main method. Up to now you were forced to copy&paste some form of the static main method to every tool.

Reply:
I am not sure this would be helpful to the reader. The feature was designed to simplify the mental model of beginners, not to save a minimal step for tool builders.

Reply:
Sure, this is only my limited view of a non-beginner :)